usermanager/src/test/java/org/pesmypetcare/usermanager/clients/user/UserMedalManagerClientTest.java

```diff
@@ -0,0 +1,131 @@
+package org.pesmypetcare.usermanager.clients.user;
+
+import com.google.gson.Gson;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.pesmypetcare.httptools.HttpClient;
+import org.pesmypetcare.httptools.HttpParameter;
+import org.pesmypetcare.httptools.HttpResponse;
+import org.pesmypetcare.httptools.exceptions.MyPetCareException;
+import org.pesmypetcare.usermanager.BuildConfig;
+import org.pesmypetcare.usermanager.datacontainers.user.UserMedalData;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+
+/**
+ * @author Oriol Catalán
+ */
+@RunWith(PowerMockRunner.class)
+public class UserMedalManagerClientTest {
+    private static final String BASE_URL = BuildConfig.URL;
+    private static final String USERMEDAL_PATH = BASE_URL + "usermedal/";
+    private static final String USERMEDAL_PARAMETER = "usermedal";
+
+    private static String medalName;
+    private static String owner;
+    private static String token;
+    private static String medalNameEncoded;
+    private static String ownerNameEncoded;
+    private static String fieldNameEncoded;
+    private static String field;
+    private static Gson gson;
+    private static UserMedalData medal;
+    private static Map<String, String> headers;
+
+    @Mock
+    private HttpClient httpClient;
+    @Mock
+    private HttpResponse httpResponse;
+
+    @InjectMocks
+    private UserMedalManagerClient client = new UserMedalManagerClient();
+
+    @Before
+    public void setUp() {
+        gson = new Gson();
+        owner = "Rosa Melano";
+        medalName = "Walker";
+        token = "token";
+        field = "progress";
+        fieldNameEncoded = HttpParameter.encode(field);
+        ownerNameEncoded = HttpParameter.encode(owner);
+        medal = new UserMedalData("Walker",
+                new ArrayList<>(Arrays.asList(5., 10., 25., 50., 100.)),
+                "You have to walk a lot!", new byte[] {10, 55, 67, 89, 103, 116},
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['10' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5633147)